apollo-runtime/src/main/java/com/apollographql/apollo/internal/batch/BatchPoller.kt

```diff
@@ -0,0 +1,76 @@
+package com.apollographql.apollo.internal.batch
+
+import com.apollographql.apollo.api.internal.ApolloLogger
+import com.apollographql.apollo.exception.ApolloException
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.Executor
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+
+/**
+ * Class responsible for holding the queue of requests, and execute them as a batch every X ms,
+ * as defined by the BatchConfig.
+ */
+class BatchPoller(
+    private val batchConfig: BatchConfig,
+    private val dispatcher: Executor,
+    private val batchHttpCallFactory: BatchHttpCallFactory,
+    private val logger: ApolloLogger,
+    private val periodicJobScheduler: PeriodicJobScheduler = PeriodicJobSchedulerImpl()
+) {
+
+  private val queryQueue: ConcurrentLinkedQueue<QueryToBatch> = ConcurrentLinkedQueue()
```

Comment:
Since the handling of the queue is now synchronized, the queue can certainly be moved to a regular `LinkedList` ? Or `Dequeue` ? I'm not super familiar with the different Java Queues but it sounds like that'd have lower overhead?

Reply:
yeah I think `LinkedList` should be sufficient now